### PR TITLE
Tweaks to be able to compare different type analyzers

### PR DESCRIPTION
**Pull request by mamikonyana**

### CODE REVIEW

File: bench/execute_workflows.py [hunk 1]
```diff
@@ -92,13 +91,15 @@ def parse_results(results_file):
...
+
 def get_stats(value_list):
   value_list = [float(value) for value in value_list]
   mean = sum(value_list) / len(value_list)
   stddev = (sum([(value - mean)**2 for value in value_list]) / len(value_list)) ** 0.5
   return mean, stddev
 
-def run_workload(config_parameters, print_itemsets=True):
+
+def run_workload(config_parameters, repeats, print_itemsets=True):
```
Comment (deepakn94) [inline bench/execute_workflows.py]: Let's rename this to `num_runs`.

File: bench/execute_workflows.py [hunk 2]
```diff
@@ -152,8 +153,7 @@ def run_workload(config_parameters, print_itemsets=True):
   print "Mean tuples / second:", mean_tuples_per_second, ", Stddev tuples / second:", stddev_tuples_per_second
 
 
-def run_all_workloads(configurations,
-                      script_arguments={},
+def run_all_workloads(configurations, defaults, repeats_per_workload,
```
Comment (deepakn94) [inline bench/execute_workflows.py]: Here too.

File: bench/execute_workflows.py [hunk 3]
```diff
@@ -198,6 +193,11 @@ def parse_args():
                       type=argparse.FileType('r'),
                       default='conf/sweeping_parameters.json',
                       help='File with a dictionary of sweeping parameters')
+  parser.add_argument('--workload-repeats', default=5, type=int,
```
Comment (deepakn94) [inline bench/execute_workflows.py]: `num_runs`

**Comment by deepakn94** [comment]
Sorry for the nitpicks!

**Comment by mamikonyana** [comment]
it's good. I even like that!

**Comment by deepakn94** [comment]
Actually this leads to undesired behavior. We currently _want_ to not explicitly specify an `outlierDetectorType` when we run experiments, since we want to dynamically pick an outlier detection algorithm depending on the dimension of the input points.

**Comment by deepakn94** [comment]
Better behavior would be to plug in the `outlierDetectorType` into the configuration file only if `outlierDetectorType` is provided in the configuration dict.

**Comment by pbailis** [comment]
We could make specifying the `outlierDetectorType` required. I think eventually we want to. However, our scripts don't currently support it.

**Comment by deepakn94** [comment]
Yeah, I guess. It's a pain to go ahead and pick between MAD and MCD for every workload. (especially when that information can be inferred directly from the dimensionality of the input points)

**Pull request closed without merging**